curriculum/challenges/english/08-coding-interview-prep/rosetta-code/zhang-suen-thinning-algorithm.english.md

```diff
@@ -44,49 +44,51 @@ It produces the thinned output:
      #                             ############
                        ###                          ###
 
-                                                           </pre>
+</pre>
+
 <h2>Algorithm</h2>
 Assume black pixels are one and white pixels zero, and that the input image is a rectangular N by M array of ones and zeroes.
 The algorithm operates on all black pixels P1 that can have eight neighbours. The neighbours are, in order, arranged as:
-<table border="1">
-  <tr><td>P9</td><td>P2</td><td>P3</td></tr>
-  <tr><td>P8</td><td><b>P1</b></td><td>P4</td></tr>
-  <tr><td>P7</td><td>P6</td><td>P5</td></tr>
-</table>
-Obviously the boundary pixels of the image cannot have the full eight neighbours.
 
-    Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2,P3,P4,P5,P6,P7,P8,P9,P2. (Note the extra P2 at the end - it is circular).
+| P9 | P2 | P3 |
+|:--:|:--:|:--:|
+| <b>P8</b> | <b>P1</b> | <b>P4</b> |
+| <b>P7</b> | <b>P6</b> | <b>P5</b> |
 
-
-    Define $B(P1)$ = the number of black pixel neighbours of P1. ( = sum(P2 .. P9) )
+Obviously the boundary pixels of the image cannot have the full eight neighbours.
+  <ul>
+    <li>Define $A(P1)$ = the number of transitions from white to black, (0 -> 1) in the sequence P2,P3,P4,P5,P6,P7,P8,P9,P2. (Note the extra P2 at the end - it is circular).</li>
```

Comment:
Let's add some whitespace between the numbers here:

`P2, P3, P4, P5, P6, P7, P8, P9, P2.`